nameko/amqp/publish.py

```diff
@@ -15,17 +18,22 @@ class UndeliverableMessage(Exception):
 
 
 @contextmanager
-def get_connection(amqp_uri, ssl=None):
-    conn = Connection(amqp_uri, ssl=ssl)
+def get_connection(amqp_uri, ssl=None, transport_options=None):
+    if not transport_options:
+        transport_options = deepcopy(DEFAULT_TRANSPORT_OPTIONS)
+    conn = Connection(amqp_uri, transport_options=transport_options, ssl=ssl)
+
     with connections[conn].acquire(block=True) as connection:
         yield connection
 
 
 @contextmanager
-def get_producer(amqp_uri, confirms=True, ssl=None):
-    transport_options = {
-        'confirm_publish': confirms
-    }
+def get_producer(amqp_uri, confirms=True, ssl=None, transport_options=None):
+    if transport_options:
+        transport_options['confirm_publish'] = confirms
+    else:
+        transport_options = deepcopy(DEFAULT_TRANSPORT_OPTIONS)
+        transport_options['confirm_publish'] = confirms
```

Comment:
This would be slightly neater here:

``` python
if transport_options is None:
    transport_options = DEFAULT_TRANSPORT_OPTIONS.copy()
transport_options['confirm_publish'] = confirms
```

No need for deepcopy if `transport_options` doesn't contain any nested mutable objects.

I wonder if we should deprecate passing `confirms` as its own keyword argument and expect it to be passed as part of `transport_options`.